lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -4271,12 +4271,7 @@ private boolean doFlush(boolean applyAllDeletes) throws IOException {
         boolean flushSuccess = false;
         try {
           long seqNo = docWriter.flushAllThreads();
-          if (seqNo < 0) {
-            seqNo = -seqNo;
-            anyChanges = true;
-          } else {
-            anyChanges = false;
-          }
+          anyChanges = seqNo < 0;
```

Comment:
Or perhaps not have `seqNo` even i.e. inline?
```suggestion
          anyChanges = (docWriter.flushAllThreads() < 0);
```

Reply:
It makes sense, thanks @cpoerschke . I fixed it.